src/js/components/sidebar.jsx

```diff
@@ -140,6 +164,13 @@ export default class SideBar extends React.Component {
       return;
     }
 
+    // select by click feature
+    let state = this.state;
+    let tab = this.state.tabsById.get(tabId);
+    if (state.selectByClick && !tab.filtered) {
+      tab.selected = true;
```

Comment:
This mutates state without an explicit call to `setState`. React does state comparisons by checking for reference equality

Instead, you need to take a copy of `tabsById` and replace the element for the change tab with a new object. For example:

```javascript
const newTabsById = new Map(state.tabsById.entries());

newTabsById.set(tabId, Object.assign({}, tabsById.get(tabId), { selected: true }));

// Since we are calling setState below, we can coalesce it into a single call.
this.setState({
    activeTabId: tabId,
    tabsById: newTabsById,
});
```